### DISCUSSION TITLE
Required configurations for RCP build

### DISCUSSION THREAD

**Discussion by testter2** [openthread/openthread#10070] [Q&A] [answered]
Hi,

I recently had a successful build for OpenThread as FTD device (cli app) to a new testing platform, and now I want to build as RCP device so that i can integrate with raspberry pi as OTBR.

I'm uncertain about the specific configurations required for RCP build.

I have tried to follow steps from the following
https://openthread.io/guides/border-router/build#build-and-flash-rcp
https://openthread.io/codelabs/openthread-border-router#1

but still looking for the required files to be included in simple RCP build.

I am seeking guidance in modifying my FTD build application to include RCP application.

Thanks in advance

**Comment by Irving-cl** [comment]
Hi @testter2,

How did you build your FTD application? I assume you have added your new testing platform under `examples/platforms/new_testing_platform`. Then you can build your RCP by something like:
```
./script/cmake-build {your_new_testing_platform} -DOT_APP_CLI=OFF -DOT_APP_NCP=OFF -DOT_FTD=OFF -DOT_MTD=OFF
```
This [doc](https://github.com/openthread/openthread/tree/main/src/posix) may help. The key configuration for RCP device is `OT_RCP`. The build command here disables `OT_FTD` and `OT_MTD` to save time. If you also want to build them, you can set them as ON. 

Note that you will also need to update the build script to add the name of your new platform: https://github.com/openthread/openthread/blob/main/script/cmake-build#L70

And of course, you'll need to add cmake files to build the sources of your platform.

> I am seeking guidance in modifying my FTD build application to include RCP application.

BTW, not sure if I understand this, an application can be either FTD or RCP. But cannot be both.

**Comment by testter2** [comment]
Hi @Irving-cl 

my application follows a different structure from the one outlined in https://openthread.io/guides/porting/set-up-the-build-environment#repository_structure

I have built FTD App with direct use of Makefiles, So I want to modify the app so that it builds for RCP only not FTD.

I plan to take the following steps:
1- removing unnecessary directories from build.
2- Incorporate the necessary macro definitions, such as OT_RCP, into the build process to ensure that the application is built exclusively for RCP.

Considering these adjustments, I kindly request guidance on the necessary configuration and files essential for RCP build

**Reply by jwhui** [reply, marked as answer]
Given that you are using your on build system, take a look at the CMake files, especially around the `OT_RCP` option.

**Reply by testter2** [reply]
Thanks for your reply
